tests/test_unwrap.py

```diff
@@ -119,28 +122,59 @@ def test_unwrap_snaphu_nodata(self, tmp_path, list_of_gtiff_ifgs, corr_raster):
     def test_goldstein(self, tmp_path, list_of_gtiff_ifgs, corr_raster, method):
         # test other init_method
         unw_filename = tmp_path / "unwrapped.unw.tif"
+        scratch_dir = tmp_path / "scratch"
+        scratch_dir.mkdir()
         unw_path, conncomp_path = dolphin.unwrap.unwrap(
             ifg_filename=list_of_gtiff_ifgs[0],
             corr_filename=corr_raster,
             unw_filename=unw_filename,
             nlooks=1,
             unwrap_method=method,
             run_goldstein=True,
+            scratchdir=scratch_dir,
         )
         assert unw_path.exists()
         assert conncomp_path.exists()
 
+    def test_interp_loop(self, list_of_gtiff_ifgs, corr_raster):
+        ifg = io.load_gdal(list_of_gtiff_ifgs[0])
+        corr = io.load_gdal(corr_raster)
+        interpolated_ifg = np.zeros((100, 200), dtype=np.complex64)
+        indices = np.array(
+            dolphin.similarity.get_circle_idxs(
+                max_radius=51, min_radius=0, sort_output=False
+            )
+        )
+        dolphin.interpolation._interp_loop(
+            ifg,
+            corr,
+            weight_cutoff=0.5,
+            num_neighbors=20,
+            alpha=0.75,
+            indices=indices,
+            interpolated_ifg=interpolated_ifg,
+        )
+        assert math.isclose(
+            np.angle(interpolated_ifg[0, 100]), 0.73608005, rel_tol=0.0000001
+        )
+        assert math.isclose(
+            np.angle(interpolated_ifg[70, 101]), 0.56214905, rel_tol=0.0000001
+        )
```

Comment:
Glad to see some numerical check of the interpolation function. Would you please explain how do you expect the value to be 0.73608005 and 0.56214905 at those two selected pixels after interpolation?

Reply:
I ran the function locally and checked the results, since the interferogram has some fixed real values, the interpolation should be the same.

Reply:
If I understand correctly the test is working with an interferogram that has a constant phase (same phase at all pixels). So you expect the interpolation will lead to the same constant phase. Is that correct?

Reply:
The sample interferogram looks like the attached figure, and I am interpolating for the middle masked line, I expect the interpolation would always lead to the same phase values

<img width="415" alt="Screenshot 2024-03-18 at 3 15 25 PM" src="https://github.com/isce-framework/dolphin/assets/37273875/daca18d3-a007-4ed2-8a5c-3d87241ad6f1">

Reply:
Thanks @mirzaees for the plot. Can you elaborate what do you mean by the "Same phase value"?